funcs.go

```diff
@@ -15,6 +15,10 @@ import (
 //	logger.LogCtx(ctx, "level", "error", "message", "failed to do a thing", "stack", microerror.Stack(err))
 //
 func Stack(err error) string {
+	if err == nil {
+		return "<nil>"
```

Comment:
I think zero value would be better here. Any reason not to go with it?

Reply:
I think when you see json like `"stack": "<nil>"` it is more reassuring that this isn't a bug in the printing code.

Reply:
And `<nil>` is taken from `fmt.Printf("%v\n", nil)` output  https://play.golang.org/p/h-4E8grakeS

Reply:
Ok then we could just add the nil case below. 

```
case nil:
	fmt.Printf("%v", nil)
```

Reply:
It won't work because this is type assertion switch but I can do:
```suggestion
		return fmt.Printf("%v", nil)
```

Reply:
It works. I tried it out.

Reply:
TIL. Changed. I also added test.

Reply:
I didn't knew that either but I was curious and saw an opportunity that then validated. 🎉